### PR TITLE
fix(file-uploader): add default for role prop, update tests & story

### LINKED ISSUE #3316: AVT 1 - React FileUploaderButton has a DAP violation
## Environment
macOS Mojave version 10.14.5
Version 75.0.3770.100 (Official Build) (64-bit)
Carbon v10 - React

## Detailed Description
Run DAP on any of the FileUpLoaderButton and the following DAP violation is present:
Screen Shot 2019-07-09 at 1 45 04 PM

Note: To fix this issue remove "role" from the label.

### PR DESCRIPTION
**Pull request by jendowns**
Closes #3316 

#### Changelog

**New**

- add default for `role` prop

**Changed**

- add `button` as a default knob value in story
- change test to check for presence of default `role` prop (versus before, it was checking that there was no default `role`)

**Removed**

- remove duplicate `role` attribute

### CODE REVIEW

File: packages/react/src/components/FileUploader/FileUploader.js [hunk 1]
```diff
@@ -159,7 +160,6 @@ export class FileUploaderButton extends Component {
     return (
       <>
         <label
-          role="button"
```
Comment (jendowns) [inline packages/react/src/components/FileUploader/FileUploader.js:162]: Note that this is already defined on line 171

**Review by joshblack — APPROVED** [review]

**Comment by joshblack** [comment]
Quick question, what's the use-case for configuring the role?

**Comment by jendowns** [comment]
@joshblack My opinion is that if the `role` is required for accessibility purposes, then it should at least have an acceptable `defaultProp` (Another option is of course to make `role` a required prop, but that would be considered a breaking change right?)

And then for the story, similar thing -- it's nice that there's a knob there where a user can change the `role` value, but it should have something by default there (instead of set to an empty string) so that it doesn't set off a DAP violation. Also IMO it's nice to demonstrate a good practice there.

**Comment by joshblack** [comment]
@jendowns sorry, the diff totally confused me lol. Didn't realize `role` was being set twice!

**Review by tw15egan — APPROVED** [review]
Thanks for fixing that, @jendowns! 👍 🎉

**Comment by snidersd** [comment]
Retested this issue and there is a new DAP error (See screenshot below) because role="button" is on the label. [See ARIA authoring practices for a button for examples](https://www.w3.org/TR/wai-aria-practices/examples/button/button.html).
Screen Shot 2019-07-16 at 12 00 02 PM
FYI... there's a new ruleset out for July 2019 that should be used. 
Screen Shot 2019-07-16 at 12 19 15 PM

**Pull request merged**